stubs/ldap3/ldap3/utils/asn1.pyi

```diff
@@ -1,37 +1,49 @@
-from typing import Any
-from typing_extensions import TypeAlias
+from _typeshed import Incomplete, IndexableBuffer, SliceableBuffer
+from collections.abc import Callable, Mapping
+from typing import Any, TypeVar, overload
+from typing_extensions import Final, TypeAlias
 
-# Enable when pyasn1 gets stubs:
-# from pyasn1.codec.ber.encoder import AbstractItemEncoder
-_AbstractItemEncoder: TypeAlias = Any
+from pyasn1.codec.ber.encoder import AbstractItemEncoder, tagMap, typeMap
 
-CLASSES: Any
+# Use _typeshed.SupportsGetItemBuffer after PEP 688
+SupportsGetItemBuffer: TypeAlias = SliceableBuffer | IndexableBuffer
+_Unused: TypeAlias = object
+_R = TypeVar("_R")
+_B = TypeVar("_B", bound=SupportsGetItemBuffer)
+# The possible return type is a union of all other decode methods, ie: AnyOf[Incomplete | bool]
+_AllDecodersReturnType: TypeAlias = Any
 
-class LDAPBooleanEncoder(_AbstractItemEncoder):
+CLASSES: Final[dict[tuple[bool, bool], int]]
+
+class LDAPBooleanEncoder(AbstractItemEncoder):
     supportIndefLenMode: bool
     # Requires pyasn1 > 0.3.7
-    def encodeValue(self, value, asn1Spec, encodeFun, **options): ...
+    def encodeValue(self, value: bool | int, asn1Spec: _Unused, encodeFun: _Unused, **options: _Unused): ...
 
-customTagMap: Any
-customTypeMap: Any
+customTagMap = tagMap
+customTypeMap = typeMap
```

Comment:
Hmm, maybe we should just remove these from the stub? They look like artefacts of stubgen-generated stubs.

Looking at the source code, they seem pretty implementation-detail-y: the block of code is introduced with [a comment saying](https://github.com/cannatag/ldap3/blob/040e09fe37cf06801960d494dfadf0a6ed94e38f/ldap3/utils/asn1.py#L39-L40):

```
# Monkeypatching of pyasn1 for encoding Boolean with the value 0xFF for TRUE
# THIS IS NOT PART OF THE FAST BER DECODER
```

I say we remove these symbols from the stub, and add an allowlist entry (with a comment saying why they're omitted). Then we don't have to add implementation details to the `pyasn1` stubs. (All of the `tagMap` and `typeMap` symbols are marked with comments in the source code saying that they're deprecated, and none of them are included in `__all__` at runtime).

Reply:
Yeah alright. Sounds good to me. I didn't realize they were marked as deprecated.